apps/ideaspace/src/components/common/DialogBox/DialogBox.js

```diff
@@ -0,0 +1,108 @@
+import Modal from 'react-modal';
+import { useState } from 'react';
+import {
+  Dialog, DialogActions,
+  DialogContent, DialogContentText, DialogTitle,
+} from '@mui/material';
+import IconButton from '@mui/material/IconButton';
+import { atoms } from '@devlaunchers/components/src/components';
+
+const useConfirm = (title, message) => {
+  const [promise, setPromise] = useState(null);
+
+  const confirm = () => new Promise((resolve, reject) => {
+    setPromise({ resolve });
+  });
+
+  const handleClose = () => {
+    setPromise(null);
+  };
+
+  const handleConfirm = () => {
+    promise?.resolve(true);
+    handleClose();
+  };
+
+  const handleCancel = () => {
+    promise?.resolve(false);
+    handleClose();
+  };
+
+  const ConfirmationDialog = () => (
+
+    <Dialog
+      open={promise !== null}
+      fullWidth
+      sx={{
+        '& .MuiDialog-paper': {
+          borderRadius: '1rem',
+          width: '25rem',
+        },
+      }}
+    >
+      <IconButton
+        aria-label="close"
+        onClick={handleCancel}
+        sx={{
+          position: 'absolute',
+          right: '1rem',
+          top: '0.8rem',
+          backgroundColor: 'grey',
+          borderRadius: 0,
+          height: '1.5rem',
+          width: '1.5rem',
+        }}
+      >
+        x
+      </IconButton>
+
+      <DialogTitle
+        sx={{
+          paddingTop: '2.2rem',
+          paddingBottom: '0.5rem'
+        }}
+      >
+        {title}
+      </DialogTitle>
+
+      <DialogContent>
+        <DialogContentText
+          sx={{
+            fontSize: '1.1rem',
+            paddingBottom: '0.5rem',
```

Comment:
Can we avoid using in-line styles here? It looks like this is shared with the `DialogContentText` in the `NoticeBox`, so we should only need to define the styling once in a styled component.

Reply:
<DialogContentText> is from @mui/material', can it be redefined with new style?